zcash_test_vectors/transaction.py

```diff
@@ -471,7 +479,11 @@ def __init__(self, rand, consensus_branch_id):
             if is_coinbase:
                 # set enableSpendsOrchard = 0
                 self.flagsOrchard &= 2
-            self.valueBalanceOrchard = rand.u64() % (MAX_MONEY + 1)
+            valueBalanceRand = rand.u64()
+            if valueBalanceRand & 0x8000000000000000:
+                self.valueBalanceOrchard = (valueBalanceRand % (MAX_MONEY + 1)) - MAX_MONEY
+            else:
+                self.valueBalanceOrchard = valueBalanceRand % (MAX_MONEY + 1)
             self.anchorOrchard = PallasBase(leos2ip(rand.b(32)))
             self.proofsOrchard = rand.b(rand.u8() + 32) # Proof will always contain at least one element
             self.bindingSigOrchard = RedPallasSignature(rand)
```

Comment:
Below here: `version_bytes` is confusingly named because it doesn't return `bytes`.